pkgs/servers/web-apps/funkwhale/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchurl, unzip }:
+
+# Look for the correct urls for build_front and build_api artifacts on the tags page of the project : https://dev.funkwhale.audio/funkwhale/funkwhale/pipelines?scope=tags
+# Attention : do not use the url "https://dev.funkwhale.audio/funkwhale/funkwhale/-/jobs/artifacts/${release}/download?job=" : it is not guaranteed to be stable
+
+let
+  release = "0.20.0";
+  srcs = {
+    api = fetchurl {
+      url = https://dev.funkwhale.audio/funkwhale/funkwhale/-/jobs/31311/artifacts/download;
+      name =  "api.zip";
+      sha256 = "1v8v5rha21ksdqnj73qkvc35mxal82hypxa5gnf82y1cjk2lp4w7";
+    };
+    frontend = fetchurl {
+      url = https://dev.funkwhale.audio/funkwhale/funkwhale/-/jobs/31308/artifacts/download;
+      name =  "frontend.zip";
+      sha256 = "02pc6j83sn5l8wz7i2r649pff3gs5021isx9d5l9xsb5cndkq0b4";
+    };
+  };
+in stdenv.mkDerivation {
+  name = "funkwhale";
+  version = "${release}";
+  src = srcs.api;
+  nativeBuildInputs = [ unzip ];
+  postPatch = ''
+    substituteInPlace requirements/base.txt \
+      --replace "django-cleanup==3.2.0" django-cleanup
+  '';
+
+  installPhase = ''
+    mkdir $out
+    cp -R ./* $out
+    unzip ${srcs.frontend} -d $out
```

Comment:
In nixpkgs there's a structure for $out you need to follow. And since this is a python package, you should use `buildPythonPackage` to build, which will do a lot of things automatically. See section 10.15.2.2.2 in the [Nixpkgs manual](https://nixos.org/nixpkgs/manual/#building-packages-and-applications). This way you can also specify the dependencies in the package directly.

Reply:
This is a Django application and they usually do not provide any installer. While it would be nice if funkwhale would be usable without the NixOS module, given its a Django app it is typically managed using the `manage.py` script and various other executables are executed (gunicorn e.g.). Therefore, I think this approach is alright.

Note in another PR (https://github.com/NixOS/nixpkgs/pull/67951/files#diff-5e9f694ce07aa22a80eb0c049c03beaf) I recommended putting the contents of the Django application in a `site-packages` folder using `buildPythonPackage` because then the application could be added to the env (that would correspond to adding the `funkwhale` derivation to the Python env. That would be even nicer, especially if the Python env would be moved here!